Missing Eslint Rules

Missing some rules causing a build error.

Error: Definition for rule '@typescript-eslint/comma-dangle' was not found.  @typescript-eslint/comma-dangle
Error: Definition for rule '@typescript-eslint/semi' was not found.  @typescript-eslint/semi

https://console.cloud.google.com/cloud-build/builds/a766fe79-ee6d-4589-907a-cd8e31d6b716?project=pagoda-data-stack-dev

Step #0: 
ERROR: build step 0 "gcr.io/cloud-builders/docker" failed: step exited with non-zero status: 1
ERROR
Finished Step #0
Step #0: The command '/bin/sh -c npm run build' returned a non-zero code: 1
Step #0: npm notice 
Step #0: npm notice Run `npm install -g npm@10.8.1` to update!
Step #0: npm notice Changelog: <https://github.com/npm/cli/releases/tag/v10.8.1>
Step #0: npm notice New major version of npm available! 8.19.4 -> 10.8.1
Step #0: npm notice 
Step #0: info  - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/basic-features/eslint#disabling-rules
Step #0: 
Step #0: 1:1  Error: Definition for rule '@typescript-eslint/comma-dangle' was not found.  @typescript-eslint/comma-dangle
Step #0: 1:1  Error: Definition for rule '@typescript-eslint/semi' was not found.  @typescript-eslint/semi
Step #0: ./src/utils/helpers.ts
Step #0: 
Step #0: 1:1  Error: Definition for rule '@typescript-eslint/comma-dangle' was not found.  @typescript-eslint/comma-dangle
Step #0: 1:1  Error: Definition for rule '@typescript-eslint/semi' was not found.  @typescript-eslint/semi
Step #0: ./src/utils/formatTimestamp.ts
Step #0: 
Step #0: 1:1  Error: Definition for rule '@typescript-eslint/comma-dangle' was not found.  @typescript-eslint/comma-dangle
Step #0: 1:1  Error: Definition for rule '@typescript-eslint/semi' was not found.  @typescript-eslint/semi
Step #0: ./src/utils/calculateTimestamp.ts
Step #0: 
Step #0: 1:1  Error: Definition for rule '@typescript-eslint/comma-dangle' was not found.  @typescript-eslint/comma-dangle
Step #0: 1:1  Error: Definition for rule '@typescript-eslint/semi' was not found.  @typescript-eslint/semi
Step #0: ./src/constants/DurationMap.ts
Step #0: 
Step #0: 1:1  Error: Definition for rule '@typescript-eslint/comma-dangle' was not found.  @typescript-eslint/comma-dangle
Step #0: 1:1  Error: Definition for rule '@typescript-eslint/semi' was not found.  @typescript-eslint/semi
Step #0: ./src/components/Logs/GraphQL/QueryValidation.ts
Step #0: 
Step #0: 1:1  Error: Definition for rule '@typescript-eslint/comma-dangle' was not found.  @typescript-eslint/comma-dangle
Step #0: 1:1  Error: Definition for rule '@typescript-eslint/semi' was not found.  @typescript-eslint/semi
Step #0: ./src/components/Logs/GraphQL/Query.ts
Step #0: 
Step #0: Failed to compile.
Step #0: 
Step #0:  ⚠ The Next.js plugin was not detected in your ESLint configuration. See https://nextjs.org/docs/basic-features/eslint#migrating-existing-config
Step #0: 
Step #0:    Linting and checking validity of types ...
Step #0: 
Step #0: https://nextjs.org/telemetry
Step #0: You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
Step #0: This information is used to shape Next.js' roadmap and prioritize features.
Step #0: Attention: Next.js now collects completely anonymous telemetry regarding usage.
Step #0: 
Step #0: > next build
Step #0: > query-api-react@0.1.0 build
Step #0: 
Step #0:  ---> Running in d8aa035bbbad
Step #0: Step 13/21 : RUN npm run build
Step #0:  ---> d3d2928d92fc
Step #0: Step 12/21 : COPY --from=dependencies ./node_modules ./node_modules
Step #0:  ---> 1fccbe7448d7
Step #0: Step 11/21 : COPY . .
Step #0:  ---> f9603c00e419
Step #0: Removing intermediate container 95b8bee7b53f
Step #0:  ---> Running in 95b8bee7b53f
Step #0: Step 10/21 : WORKDIR ./
Step #0:  ---> 4d8cf2a71525
Step #0: Removing intermediate container 0d4daa7f3004
Step #0:  ---> Running in 0d4daa7f3004
Step #0: Step 9/21 : ENV NEXT_PUBLIC_REGISTRY_CONTRACT_ID=$NEXT_PUBLIC_REGISTRY_CONTRACT_ID
Step #0:  ---> e756ae22817a
Step #0: Removing intermediate container 14be79ad94a4
Step #0:  ---> Running in 14be79ad94a4
Step #0: Step 8/21 : ENV NEXT_PUBLIC_HASURA_ENDPOINT=$NEXT_PUBLIC_HASURA_ENDPOINT
Step #0:  ---> d99016ae1458
Step #0: Removing intermediate container 98ed1aeaef8e
Step #0:  ---> Running in 98ed1aeaef8e
Step #0: Step 7/21 : ARG NEXT_PUBLIC_HASURA_ENDPOINT
Step #0:  ---> f90c88fa713b
Step #0: Removing intermediate container 7624aba2a98a
Step #0:  ---> Running in 7624aba2a98a
Step #0: Step 6/21 : ARG NEXT_PUBLIC_REGISTRY_CONTRACT_ID
Step #0:  ---> 2573171e0124
Step #0: Step 5/21 : FROM node:16-alpine as builder
Step #0:  ---> 8bb08c7b9948
Step #0: Removing intermediate container 50fd86a1c2ee
Step #0: npm notice 
Step #0: npm notice Run `npm install -g npm@10.8.1` to update!
Step #0: npm notice Changelog: <https://github.com/npm/cli/releases/tag/v10.8.1>
Step #0: npm notice New major version of npm available! 8.19.4 -> 10.8.1
Step #0: npm notice 
Step #0: Run `npm audit` for details.
Step #0: 
Step #0:   npm audit fix --force
Step #0: To address all issues (including breaking changes), run:
Step #0: 
Step #0: 1 high severity vulnerability
Step #0: 
Step #0:   run `npm fund` for details
Step #0: 183 packages are looking for funding
Step #0: 
Step #0: added 1083 packages, and audited 1084 packages in 32s
Step #0: 
Step #0: npm WARN deprecated glob@7.1.7: Glob versions prior to v9 are no longer supported
Step #0: npm WARN deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
Step #0: npm WARN deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
Step #0: npm WARN EBADENGINE }
Step #0: npm WARN EBADENGINE   current: { node: 'v16.20.2', npm: '8.19.4' }
Step #0: npm WARN EBADENGINE   required: { node: '^18.18.0 || >=20.0.0' },
Step #0: npm WARN EBADENGINE   package: '@typescript-eslint/typescript-estree@7.12.0',
Step #0: npm WARN EBADENGINE Unsupported engine {
Step #0: npm WARN EBADENGINE }
Step #0: npm WARN EBADENGINE   current: { node: 'v16.20.2', npm: '8.19.4' }
Step #0: npm WARN EBADENGINE   required: { node: '^18.18.0 || >=20.0.0' },
Step #0: npm WARN EBADENGINE   package: '@typescript-eslint/types@7.12.0',
Step #0: npm WARN EBADENGINE Unsupported engine {
Step #0: npm WARN EBADENGINE }
Step #0: npm WARN EBADENGINE   current: { node: 'v16.20.2', npm: '8.19.4' }
Step #0: npm WARN EBADENGINE   required: { node: '^18.18.0 || >=20.0.0' },
Step #0: npm WARN EBADENGINE   package: '@typescript-eslint/visitor-keys@7.12.0',
Step #0: npm WARN EBADENGINE Unsupported engine {
Step #0: npm WARN EBADENGINE }
Step #0: npm WARN EBADENGINE   current: { node: 'v16.20.2', npm: '8.19.4' }
Step #0: npm WARN EBADENGINE   required: { node: '^18.18.0 || >=20.0.0' },
Step #0: npm WARN EBADENGINE   package: '@typescript-eslint/utils@7.12.0',
Step #0: npm WARN EBADENGINE Unsupported engine {
Step #0: npm WARN EBADENGINE }
Step #0: npm WARN EBADENGINE   current: { node: 'v16.20.2', npm: '8.19.4' }
Step #0: npm WARN EBADENGINE   required: { node: '^18.18.0 || >=20.0.0' },
Step #0: npm WARN EBADENGINE   package: '@typescript-eslint/type-utils@7.12.0',
Step #0: npm WARN EBADENGINE Unsupported engine {
Step #0: npm WARN EBADENGINE }
Step #0: npm WARN EBADENGINE   current: { node: 'v16.20.2', npm: '8.19.4' }
Step #0: npm WARN EBADENGINE   required: { node: '^18.18.0 || >=20.0.0' },
Step #0: npm WARN EBADENGINE   package: '@typescript-eslint/scope-manager@7.12.0',
Step #0: npm WARN EBADENGINE Unsupported engine {
Step #0: npm WARN EBADENGINE }
Step #0: npm WARN EBADENGINE   current: { node: 'v16.20.2', npm: '8.19.4' }
Step #0: npm WARN EBADENGINE   required: { node: '^18.18.0 || >=20.0.0' },
Step #0: npm WARN EBADENGINE   package: '@typescript-eslint/parser@7.12.0',
Step #0: npm WARN EBADENGINE Unsupported engine {
Step #0: npm WARN EBADENGINE }
Step #0: npm WARN EBADENGINE   current: { node: 'v16.20.2', npm: '8.19.4' }
Step #0: npm WARN EBADENGINE   required: { node: '^18.18.0 || >=20.0.0' },
Step #0: npm WARN EBADENGINE   package: '@typescript-eslint/eslint-plugin@7.12.0',
Step #0: npm WARN EBADENGINE Unsupported engine {
Step #0: npm WARN     node_modules/@graphiql/plugin-code-exporter
Step #0: npm WARN     graphiql-code-exporter@"^3.0.3" from @graphiql/plugin-code-exporter@0.3.5
Step #0: npm WARN   node_modules/@graphiql/plugin-code-exporter/node_modules/graphiql-code-exporter
Step #0: npm WARN   peer graphql@"^14.1.1" from graphiql-code-exporter@3.0.3
Step #0: npm WARN node_modules/graphql
Step #0: npm WARN Conflicting peer dependency: graphql@14.7.0
Step #0: npm WARN 
Step #0: npm WARN   node_modules/@graphiql/plugin-code-exporter
Step #0: npm WARN   graphiql-code-exporter@"^3.0.3" from @graphiql/plugin-code-exporter@0.3.5
Step #0: npm WARN node_modules/@graphiql/plugin-code-exporter/node_modules/graphiql-code-exporter
Step #0: npm WARN peer graphql@"^14.1.1" from graphiql-code-exporter@3.0.3
Step #0: npm WARN Could not resolve dependency:
Step #0: npm WARN 
Step #0: npm WARN     node_modules/@graphiql/plugin-explorer
Step #0: npm WARN     graphiql-explorer@"^0.9.0" from @graphiql/plugin-explorer@0.3.5
Step #0: npm WARN   node_modules/@graphiql/plugin-explorer/node_modules/graphiql-explorer
Step #0: npm WARN   peer graphql@"^0.6.0 || ^0.7.0 || ^0.8.0-b || ^0.9.0 || ^0.10.0 || ^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0 || ^15.0.0" from graphiql-explorer@0.9.0
Step #0: npm WARN node_modules/graphql
Step #0: npm WARN Found: graphql@16.8.1
Step #0: npm WARN While resolving: graphiql-code-exporter@3.0.3
Step #0: npm WARN ERESOLVE overriding peer dependency
Step #0: npm WARN ERESOLVE overriding peer dependency
Step #0: npm WARN ERESOLVE overriding peer dependency
Step #0: npm WARN ERESOLVE overriding peer dependency
Step #0: npm WARN ERESOLVE overriding peer dependency
Step #0: npm WARN ERESOLVE overriding peer dependency
Step #0:  ---> Running in 50fd86a1c2ee
Step #0: Step 4/21 : RUN npm install
Step #0:  ---> 09f29e7a602c
Step #0: Step 3/21 : COPY package.json yarn.lock ./
Step #0:  ---> 4396ac5b3d80
Step #0: Removing intermediate container 01379135be05
Step #0:  ---> Running in 01379135be05
Step #0: Step 2/21 : WORKDIR ./
Step #0:  ---> 2573171e0124
Step #0: Status: Downloaded newer image for node:16-alpine
Step #0: Digest: sha256:a1f9d027912b58a7c75be7716c97cfbc6d3099f3a97ed84aa490be9dee20e787
Step #0: d9059661ce70: Pull complete
Step #0: 93b3025fe103: Pull complete
Step #0: eee371b9ce3f: Pull complete
Step #0: eee371b9ce3f: Download complete
Step #0: eee371b9ce3f: Verifying Checksum
Step #0: d9059661ce70: Download complete
Step #0: d9059661ce70: Verifying Checksum
Step #0: 93b3025fe103: Download complete
Step #0: 93b3025fe103: Verifying Checksum
Step #0: d9059661ce70: Pulling fs layer
Step #0: 93b3025fe103: Pulling fs layer
Step #0: eee371b9ce3f: Pulling fs layer
Step #0: 7264a8db6415: Already exists
Step #0: 16-alpine: Pulling from library/node
Step #0: Step 1/21 : FROM node:16-alpine as dependencies

Step #0: Sending build context to Docker daemon  1.233MB
Step #0: Already have image (with digest): gcr.io/cloud-builders/docker
Starting Step #0
BUILD
HEAD is now at 9e723b2 upgraded eslint to ^8.56.0  (#765)
 * branch            9e723b25ecd09607509f4290e1a459e5845ca9d1 -> FETCH_HEAD
From https://github.com/near/queryapi
Initialized empty Git repository in /workspace/.git/
hint: 	git branch -m <name>
hint:
hint: 'development'. The just-created branch can be renamed via this command:
hint: Names commonly chosen instead of 'master' are 'main', 'trunk' and
hint:
hint: 	git config --global init.defaultBranch <name>
hint:
hint: of your new repositories, which will suppress this warning, call:
hint: is subject to change. To configure the initial branch name to use in all
hint: Using 'master' as the name for the initial branch. This default branch name
FETCHSOURCE

starting build "a766fe79-ee6d-4589-907a-cd8e31d6b716"

--- a/frontend/.eslintrc.json
+++ b/frontend/.eslintrc.json
@@ -26,8 +26,8 @@
         "project": "./tsconfig.json"
       },
       "rules": {
-        "@typescript-eslint/semi": ["error", "always"],
-        "@typescript-eslint/comma-dangle": ["error", "only-multiline"],
+        // "@typescript-eslint/semi": ["error", "always"],
+        // "@typescript-eslint/comma-dangle": ["error", "only-multiline"],
         "@typescript-eslint/strict-boolean-expressions": "off"
       }
     }

--- a/frontend/src/utils/calculateTimestamp.ts
+++ b/frontend/src/utils/calculateTimestamp.ts
@@ -1,4 +1,4 @@
-import { DURATION_MAP } from '../constants/DurationMap'
+import { DURATION_MAP } from '../constants/DurationMap';
 
 export const calculateTimestamp = (selectedOption: string): string => {
     const currentTime: number = Date.now();
@@ -9,4 +9,4 @@ export const calculateTimestamp = (selectedOption: string): string => {
         console.log('invalid option');
         return "";
     }
-}
+};

--- a/frontend/src/utils/formatTimestamp.ts
+++ b/frontend/src/utils/formatTimestamp.ts
@@ -12,4 +12,4 @@ export const formatTimestamp = (timestamp: string): string => {
     const milliseconds = String(date.getMilliseconds()).padStart(3, '0');
 
     return `${year}-${month}-${day} ${hours}:${minutes}:${seconds}.${milliseconds}`;
-}
+};

